### PR TITLE
apply async to functions

### PR DESCRIPTION
**Pull request by LilSpazJoekp**
This is a generated tree of what needs to be converted to async:
- [x] config.py
    - [x] class: _NotSet
        - [x] \_\_bool\_\_()
        - [x] \_\_str\_\_()
    - [x] class: Config
        - [x] _config_boolean()
        - [x] _load_config()
        - [x] short_url()
        - [x] \_\_init\_\_()
        - [x] _fetch()
        - [x] _fetch_default()
        - [x] _fetch_or_not_set()
        - [x] _initialize_attributes()
- [x] objector.py
    - [x] class: Objector
        - [x] parse_error()
        - [x] check_error()
        - [x] \_\_init\_\_()
        - [x] _objectify_dict()
        - [x] objectify()
- [x] \_\_init\_\_.py
- [x] exceptions.py
    - [x] class: PRAWException
    - [x] class: APIException
        - [x] \_\_init\_\_()
    - [x] class: ClientException
    - [x] class: DuplicateReplaceException
        - [x] \_\_init\_\_()
    - [x] class: InvalidImplicitAuth
        - [x] \_\_init\_\_()
    - [x] class: InvalidURL
        - [x] \_\_init\_\_()
    - [x] class: MissingRequiredAttributeException
    - [x] class: WebSocketException
        - [x] \_\_init\_\_()
- [ ] reddit.py
    - [ ] class: Reddit
        - [x] _next_unique()
        - [x] read_only()
        - [x] read_only()
        - [x] \_\_enter\_\_()
        - [x] \_\_exit\_\_()
        - [x] \_\_init\_\_()
        - [ ] _check_for_update()
        - [x] _prepare_objector()
        - [x] _prepare_prawcore()
        - [x] _prepare_trusted_prawcore()
        - [x] _prepare_untrusted_prawcore()
        - [x] comment()
        - [x] domain()
        - [x] info()
        - [x] get()
        - [x] patch()
        - [x] post()
        - [x] put()
        - [x] random_subreddit()
        - [x] redditor()
        - [x] request()
        - [x] submission()
- [x] endpoints.py
- [x] const.py
- [ ] util/cache.py
    - [ ] class: cachedproperty
        - [ ] \_\_init\_\_()
        - [ ] \_\_get\_\_()
        - [ ] \_\_repr\_\_()
- [x] util/\_\_init\_\_.py
- [x] util/snake.py
    - [x] camel_to_snake()
    - [x] snake_case_keys()
- [x] models/redditors.py
    - [x] class: PartialRedditor
    - [x] class: Redditors
        - [x] new()
        - [x] popular()
        - [x] search()
        - [x] stream()
        - [x] partial_redditors()
- [x] models/auth.py
    - [x] class: Auth
        - [x] limits()
        - [x] implicit()
        - [x] url()
        - [x] authorize()
        - [x] scopes()
- [x] models/subreddits.py
    - [x] class: Subreddits
        - [x] _to_list()
        - [x] default()
        - [x] gold()
        - [x] new()
        - [x] popular()
        - [x] recommended()
        - [x] search()
        - [x] search_by_name()
        - [x] search_by_topic()
        - [x] stream()
- [x] models/user.py
    - [x] class: User
        - [x] preferences()
        - [x] \_\_init\_\_()
        - [x] blocked()
        - [x] contributor_subreddits()
        - [x] friends()
        - [x] karma()
        - [x] moderator_subreddits()
        - [x] multireddits()
        - [x] subreddits()
        - [x] me()
- [x] models/util.py
    - [x] permissions_string()
    - [x] stream_generator()
    - [x] class: BoundedSet
        - [x] \_\_init\_\_()
        - [x] \_\_contains\_\_()
        - [x] add()
    - [x] class: ExponentialCounter
        - [x] \_\_init\_\_()
        - [x] counter()
        - [x] reset()
- [x] models/preferences.py
    - [x] class: Preferences
        - [x] \_\_call\_\_()
        - [x] \_\_init\_\_()
        - [x] update()
- [x] models/\_\_init\_\_.py
- [ ] models/mod_action.py
    - [ ] class: ModAction
        - [ ] mod() # getter
        - [ ] mod() # setter
- [x] models/inbox.py
    - [x] class: Inbox
        - [x] all()
        - [x] collapse()
        - [x] comment_replies()
        - [x] mark_read()
        - [x] mark_unread()
        - [x] mentions()
        - [x] message()
        - [x] messages()
        - [x] sent()
        - [x] stream()
        - [x] submission_replies()
        - [x] uncollapse()
        - [x] unread()
- [x] models/comment_forest.py
    - [x] class: CommentForest
        - [x] _gather_more_comments()
        - [x] \_\_getitem\_\_()
        - [x] \_\_init\_\_()
        - [x] \_\_len\_\_()
        - [x] _insert_comment()
        - [x] _update()
        - [x] list()
        - [x] replace_more()
- [x] models/front.py
    - [x] class: Front
        - [x] \_\_init\_\_()
        - [x] best()
- [x] models/trophy.py
    - [x] class: Trophy
        - [x] \_\_init\_\_()
        - [x] \_\_str\_\_()
- [x] models/stylesheet.py
    - [x] class: Stylesheet
- [ ] models/helpers.py
    - [ ] class: LiveHelper
        - [x] \_\_call\_\_()
        - [ ] info() # does not work for me; even on regular praw
        - [x] create()
        - [x] now()
    - [x] class: MultiredditHelper
        - [x] \_\_call\_\_()
        - [x] create()
    - [x] class: SubredditHelper
        - [x] \_\_call\_\_()
        - [x] create()
- [x] models/base.py
    - [x] class: PRAWBase
        - [x] _safely_add_arguments()
        - [x] parse()
        - [x] \_\_init\_\_()
- [x] models/list/\_\_init\_\_.py
- [x] models/list/redditor.py
    - [x] class: RedditorList
- [x] models/list/trophy.py
    - [x] class: TrophyList
- [ ] models/list/base.py
    - [ ] class: BaseList
        - [x] \_\_init\_\_()
        - [x] \_\_contains\_\_()
        - [ ] \_\_getitem\_\_()
        - [ ] \_\_iter\_\_()
        - [x] \_\_len\_\_()
        - [x] \_\_str\_\_()
- [x] models/reddit/comment.py # needs further testing
    - [x] class: Comment
        - [x] id_from_url()
        - [x] _kind()
        - [x] is_root()
        - [x] mod()
        - [x] replies()
        - [x] submission()
        - [x] submission()
        - [x] \_\_init\_\_()
        - [x] \_\_setattr\_\_()
        - [x] _fetch_info()
        - [x] _fetch_data()
        - [x] _fetch()
        - [x] _extract_submission_id()
        - [x] parent()
        - [x] refresh()
    - [x] class: CommentModeration
        - [x] \_\_init\_\_()
- [ ] models/reddit/emoji.py
    - [ ] class: Emoji
        - [ ] \_\_eq\_\_()
        - [ ] \_\_hash\_\_()
        - [ ] \_\_init\_\_()
        - [ ] _fetch()
        - [ ] delete()
    - [ ] class: SubredditEmoji
        - [ ] \_\_getitem\_\_()
        - [ ] \_\_init\_\_()
        - [ ] \_\_iter\_\_()
        - [ ] add()
- [x] models/reddit/multi.py
    - [x] class: Multireddit
        - [x] sluggify()
        - [x] stream()
        - [x] \_\_init\_\_()
        - [x] _fetch_info()
        - [x] _fetch_data()
        - [x] _fetch()
        - [x] add()
        - [x] copy()
        - [x] delete()
        - [x] remove()
        - [x] update()
- [x] models/reddit/submission.py
    - [x] class: SubmissionFlair
        - [x] \_\_init\_\_()
        - [x] choices()
        - [x] select()
    - [x] class: SubmissionModeration
        - [x] \_\_init\_\_()
        - [x] contest_mode()
        - [x] flair()
        - [x] nsfw()
        - [x] set_original_content()
        - [x] sfw()
        - [x] spoiler()
        - [x] sticky()
        - [x] suggested_sort()
        - [x] unset_original_content()
        - [x] unspoiler()
    - [x] class: Submission
        - [x] id_from_url()
        - [x] _kind()
        - [x] comments()
        - [x] flair()
        - [x] mod()
        - [x] shortlink()
        - [x] \_\_init\_\_()
        - [x] \_\_setattr\_\_()
        - [x] _chunk()
        - [x] _fetch_info()
        - [x] _fetch_data()
        - [x] _fetch()
        - [x] mark_visited()
        - [x] hide()
        - [x] unhide()
        - [x] crosspost()
- [x] models/reddit/\_\_init\_\_.py
- [x] models/reddit/message.py
    - [x] class: Message
        - [x] parse()
        - [x] _kind()
        - [x] \_\_init\_\_()
        - [x] delete()
    - [x] class: SubredditMessage
        - [x] mute()
        - [x] unmute()
- [x] models/reddit/redditor.py
    - [x] class: Redditor
        - [x] from_data()
        - [x] stream()
        - [x] _kind()
        - [x] _path()
        - [x] \_\_init\_\_()
        - [x] _fetch_username()
        - [x] _fetch_info()
        - [x] _fetch_data()
        - [x] _fetch()
        - [x] _friend()
        - [x] block()
        - [x] friend()
        - [x] friend_info()
        - [x] gild()
        - [x] moderated()
        - [x] multireddits()
        - [x] trophies()
        - [x] unblock()
        - [x] unfriend()
    - [x] class: RedditorStream
        - [x] \_\_init\_\_()
        - [x] comments()
        - [x] submissions()
- [ ] models/reddit/wikipage.py
    - [ ] class: WikiPageModeration
        - [ ] \_\_init\_\_()
        - [ ] add()
        - [ ] remove()
        - [ ] settings()
        - [ ] update()
    - [ ] class: WikiPage
        - [ ] _revision_generator()
        - [ ] mod()
        - [ ] \_\_eq\_\_()
        - [ ] \_\_init\_\_()
        - [ ] \_\_repr\_\_()
        - [ ] \_\_str\_\_()
        - [ ] _fetch_info()
        - [ ] _fetch_data()
        - [ ] _fetch()
        - [ ] edit()
        - [ ] revision()
        - [ ] revisions()
- [ ] models/reddit/widgets.py
    - [ ] class: Button
    - [ ] class: Image
    - [ ] class: ImageData
    - [ ] class: MenuLink
    - [ ] class: Submenu
    - [ ] class: SubredditWidgets
        - [ ] id_card()
        - [ ] items()
        - [ ] mod()
        - [ ] moderators_widget()
        - [ ] sidebar()
        - [ ] topbar()
        - [ ] refresh()
        - [ ] \_\_getattr\_\_()
        - [ ] \_\_init\_\_()
        - [ ] \_\_repr\_\_()
        - [ ] _fetch()
    - [ ] class: SubredditWidgetsModeration
        - [ ] \_\_init\_\_()
        - [ ] _create_widget()
        - [ ] add_button_widget()
        - [ ] add_calendar()
        - [ ] add_community_list()
        - [ ] add_custom_widget()
        - [ ] add_image_widget()
        - [ ] add_menu()
        - [ ] add_post_flair_widget()
        - [ ] add_text_area()
        - [ ] reorder()
        - [ ] upload_image()
    - [ ] class: Widget
        - [ ] mod()
        - [ ] \_\_eq\_\_()
        - [ ] \_\_init\_\_()
    - [ ] class: ButtonWidget
    - [ ] class: Calendar
    - [ ] class: CommunityList
    - [ ] class: CustomWidget
        - [ ] \_\_init\_\_()
    - [ ] class: IDCard
    - [ ] class: ImageWidget
    - [ ] class: Menu
    - [ ] class: ModeratorsWidget
        - [ ] \_\_init\_\_()
    - [ ] class: PostFlairWidget
    - [ ] class: RulesWidget
        - [ ] \_\_init\_\_()
    - [ ] class: TextArea
    - [ ] class: WidgetEncoder
        - [ ] default()
    - [ ] class: WidgetModeration
        - [ ] \_\_init\_\_()
        - [ ] delete()
        - [ ] update()
- [ ] models/reddit/subreddit.py
    - [ ] class: Subreddit
        - [ ] _create_or_update()
        - [ ] _subreddit_list()
        - [ ] _kind()
        - [ ] banned()
        - [ ] collections()
        - [ ] contributor()
        - [ ] emoji()
        - [ ] filters()
        - [ ] flair()
        - [ ] mod()
        - [ ] moderator()
        - [ ] modmail()
        - [ ] muted()
        - [ ] quaran()
        - [ ] stream()
        - [ ] stylesheet()
        - [ ] widgets()
        - [ ] wiki()
        - [ ] \_\_init\_\_()
        - [ ] _fetch_info()
        - [ ] _fetch_data()
        - [ ] _fetch()
        - [ ] _submit_media()
        - [ ] _upload_media()
        - [ ] random()
        - [ ] rules()
        - [ ] search()
        - [ ] sticky()
        - [ ] submit()
        - [ ] submit_image()
        - [ ] submit_video()
        - [ ] subscribe()
        - [ ] traffic()
        - [ ] unsubscribe()
    - [ ] class: SubredditFilters
        - [ ] \_\_init\_\_()
        - [ ] \_\_iter\_\_()
        - [ ] add()
        - [ ] remove()
    - [ ] class: SubredditFlair
        - [ ] link_templates()
        - [ ] templates()
        - [ ] \_\_call\_\_()
        - [ ] \_\_init\_\_()
        - [ ] configure()
        - [ ] delete()
        - [ ] delete_all()
        - [ ] set()
        - [ ] update()
    - [ ] class: SubredditFlairTemplates
        - [ ] flair_type()
        - [ ] \_\_init\_\_()
        - [ ] _add()
        - [ ] _clear()
        - [ ] delete()
        - [ ] update()
    - [ ] class: SubredditRedditorFlairTemplates
        - [ ] \_\_iter\_\_()
        - [ ] add()
        - [ ] clear()
    - [ ] class: SubredditLinkFlairTemplates
        - [ ] \_\_iter\_\_()
        - [ ] add()
        - [ ] clear()
    - [ ] class: SubredditModeration
        - [ ] _handle_only()
        - [ ] \_\_init\_\_()
        - [ ] accept_invite()
        - [ ] edited()
        - [ ] inbox()
        - [ ] log()
        - [ ] modqueue()
        - [ ] stream()
        - [ ] removal_reasons()
        - [ ] reports()
        - [ ] settings()
        - [ ] spam()
        - [ ] unmoderated()
        - [ ] unread()
        - [ ] update()
    - [ ] class: SubredditModerationStream
        - [ ] \_\_init\_\_()
        - [ ] edited()
        - [ ] log()
        - [ ] modmail_conversations()
        - [ ] modqueue()
        - [ ] reports()
        - [ ] spam()
        - [ ] unmoderated()
        - [ ] unread()
    - [ ] class: SubredditQuarantine
        - [ ] \_\_init\_\_()
        - [ ] opt_in()
        - [ ] opt_out()
    - [ ] class: SubredditRelationship
        - [ ] \_\_call\_\_()
        - [ ] \_\_init\_\_()
        - [ ] add()
        - [ ] remove()
    - [ ] class: ContributorRelationship
        - [ ] leave()
    - [ ] class: ModeratorRelationship
        - [ ] _handle_permissions()
        - [ ] \_\_call\_\_()
        - [ ] add()
        - [ ] invite()
        - [ ] leave()
        - [ ] remove_invite()
        - [ ] update()
        - [ ] update_invite()
    - [x] class: Modmail
        - [x] \_\_call\_\_()
        - [x] \_\_init\_\_()
        - [x] _build_subreddit_list()
        - [x] bulk_read()
        - [x] conversations()
        - [x] create()
        - [x] subreddits()
        - [x] unread_count()
    - [ ] class: SubredditStream
        - [ ] \_\_init\_\_()
        - [ ] comments()
        - [ ] submissions()
    - [ ] class: SubredditStylesheet
        - [ ] \_\_call\_\_()
        - [ ] \_\_init\_\_()
        - [ ] _update_structured_styles()
        - [ ] _upload_image()
        - [ ] _upload_style_asset()
        - [ ] delete_banner()
        - [ ] delete_banner_additional_image()
        - [ ] delete_banner_hover_image()
        - [ ] delete_header()
        - [ ] delete_image()
        - [ ] delete_mobile_header()
        - [ ] delete_mobile_icon()
        - [ ] update()
        - [ ] upload()
        - [ ] upload_banner()
        - [ ] upload_banner_additional_image()
        - [ ] upload_banner_hover_image()
        - [ ] upload_header()
        - [ ] upload_mobile_header()
        - [ ] upload_mobile_icon()
    - [ ] class: SubredditWiki
        - [ ] \_\_getitem\_\_()
        - [ ] \_\_init\_\_()
        - [ ] \_\_iter\_\_()
        - [ ] create()
        - [ ] revisions()
- [x] models/reddit/more.py
    - [x] class: MoreComments
        - [x] \_\_init\_\_()
        - [x] \_\_eq\_\_()
        - [x] \_\_lt\_\_()
        - [x] \_\_repr\_\_()
        - [x] _continue_comments()
        - [x] _load_comment()
        - [x] comments()
- [x] models/reddit/collections.py
    - [x] class: Collection
        - [x] mod()
        - [x] subreddit()
        - [x] \_\_init\_\_()
        - [x] \_\_iter\_\_()
        - [x] \_\_len\_\_()
        - [x] \_\_setattr\_\_()
        - [x] _fetch_info()
        - [x] _fetch_data()
        - [x] _fetch()
        - [x] follow()
        - [x] unfollow()
    - [x] class: CollectionModeration
        - [x] _post_fullname()
        - [x] \_\_init\_\_()
        - [x] add_post()
        - [x] delete()
        - [x] remove_post()
        - [x] reorder()
        - [x] update_description()
        - [x] update_title()
    - [x] class: SubredditCollections
        - [x] mod()
        - [x] \_\_call\_\_()
        - [x] \_\_init\_\_()
        - [x] \_\_iter\_\_()
    - [x] class: SubredditCollectionsModeration
        - [x] \_\_init\_\_()
        - [x] create()
- [x] models/reddit/modmail.py
    - [x] class: ModmailConversation
        - [x] _convert_conversation_objects()
        - [x] _convert_user_summary()
        - [x] parse()
        - [x] \_\_init\_\_()
        - [x] _build_conversation_list()
        - [x] _fetch_info()
        - [x] _fetch_data()
        - [x] _fetch()
        - [x] archive()
        - [x] highlight()
        - [x] mute()
        - [x] read()
        - [x] reply()
        - [x] unarchive()
        - [x] unhighlight()
        - [x] unmute()
        - [x] unread()
    - [x] class: ModmailObject
        - [x] \_\_setattr\_\_()
    - [x] class: ModmailAction
    - [x] class: ModmailMessage
- [x] models/reddit/live.py
    - [x] class: LiveContributorRelationship
        - [x] _handle_permissions()
        - [x] \_\_call\_\_()
        - [x] \_\_init\_\_()
        - [x] accept_invite()
        - [x] invite()
        - [x] leave()
        - [x] remove()
        - [x] remove_invite()
        - [x] update()
        - [x] update_invite()
    - [x] class: LiveThread
        - [x] contrib()
        - [x] contributor()
        - [x] \_\_eq\_\_()
        - [x] \_\_getitem\_\_()
        - [x] \_\_hash\_\_()
        - [x] \_\_init\_\_()
        - [x] _fetch_info()
        - [x] _fetch_data()
        - [x] _fetch()
        - [x] discussions()
        - [x] report()
        - [x] updates()
    - [x] class: LiveThreadContribution
        - [x] \_\_init\_\_()
        - [x] add()
        - [x] close()
        - [x] update()
    - [ ] class: LiveUpdate
        - [x] contrib()
        - [x] thread()
        - [ ] \_\_init\_\_()
        - [x] \_\_setattr\_\_()
        - [x] _fetch()
    - [x] class: LiveUpdateContribution
        - [x] \_\_init\_\_()
        - [x] remove()
        - [x] strike()
- [x] models/reddit/removal_reasons.py
    - [x] class: RemovalReason
        - [x] \_\_eq\_\_()
        - [x] \_\_hash\_\_()
        - [x] \_\_init\_\_()
        - [x] _fetch()
        - [x] delete()
        - [x] update()
    - [x] class: SubredditRemovalReasons
        - [x] \_\_getitem\_\_()
        - [x] \_\_init\_\_()
        - [x] \_\_iter\_\_()
        - [x] add()
- [x] models/reddit/base.py
    - [x] class: RedditBase
        - [x] _url_parts()
        - [x] \_\_eq\_\_()
        - [x] \_\_getattr\_\_()
        - [x] \_\_hash\_\_()
        - [x] \_\_init\_\_()
        - [x] \_\_repr\_\_()
        - [x] \_\_str\_\_()
        - [x] \_\_ne\_\_()
        - [x] _fetch()
        - [x] _reset_attributes()
- [x] models/reddit/mixins/reportable.py
    - [x] class: ReportableMixin
        - [x] report()
- [x] models/reddit/mixins/editable.py
    - [x] class: EditableMixin
        - [x] delete()
        - [x] edit()
- [x] models/reddit/mixins/inboxtoggleable.py
    - [x] class: InboxToggleableMixin
        - [x] disable_inbox_replies()
        - [x] enable_inbox_replies()
- [x] models/reddit/mixins/replyable.py
    - [x] class: ReplyableMixin
        - [x] reply()
- [x] models/reddit/mixins/gildable.py
    - [x] class: GildableMixin
        - [x] gild()
- [x] models/reddit/mixins/\_\_init\_\_.py
    - [x] class: ThingModerationMixin
        - [x] _add_removal_reason()
        - [x] approve()
        - [x] distinguish()
        - [x] ignore_reports()
        - [x] lock()
        - [x] remove()
        - [x] send_removal_message()
        - [x] undistinguish()
        - [x] unignore_reports()
        - [x] unlock()
    - [x] class: UserContentMixin
- [x] models/reddit/mixins/fullname.py
    - [x] class: FullnameMixin
        - [x] fullname()
- [x] models/reddit/mixins/votable.py
    - [x] class: VotableMixin
        - [x] _vote()
        - [x] clear_vote()
        - [x] downvote()
        - [x] upvote()
- [x] models/reddit/mixins/inboxable.py
    - [x] class: InboxableMixin
        - [x] block()
        - [x] collapse()
        - [x] mark_read()
        - [x] mark_unread()
        - [x] uncollapse()
- [x] models/reddit/mixins/messageable.py
    - [x] class: MessageableMixin
        - [x] message()
- [x] models/reddit/mixins/savable.py
    - [x] class: SavableMixin
        - [x] save()
        - [x] unsave()
- [x] models/listing/domain.py
    - [x] class: DomainListing
        - [x] \_\_init\_\_()
- [x] models/listing/\_\_init\_\_.py
- [x] models/listing/listing.py
    - [x] class: Listing
        - [x] \_\_len\_\_()
        - [x] \_\_getitem\_\_()
        - [x] \_\_setattr\_\_()
    - [x] class: FlairListing
        - [x] after()
- [x] models/listing/generator.py
    - [x] class: ListingGenerator
        - [x] \_\_init\_\_()
        - [x] \_\_aiter\_\_()
        - [x] \_\_anext\_\_()
        - [x] _next_batch()
- [x] models/listing/mixins/rising.py
    - [x] class: RisingListingMixin
        - [x] random_rising()
        - [x] rising()
- [x] models/listing/mixins/submission.py
    - [x] class: SubmissionListingMixin
        - [x] duplicates()
- [x] models/listing/mixins/\_\_init\_\_.py
- [x] models/listing/mixins/redditor.py
    - [x] class: SubListing
        - [x] \_\_init\_\_()
    - [x] class: RedditorListingMixin
        - [x] comments()
        - [x] submissions()
        - [x] downvoted()
        - [x] gildings()
        - [x] hidden()
        - [x] saved()
        - [x] upvoted()
- [x] models/listing/mixins/gilded.py
    - [x] class: GildedListingMixin
        - [x] gilded()
- [x] models/listing/mixins/subreddit.py
    - [x] class: CommentHelper
        - [x] _path()
        - [x] \_\_init\_\_()
        - [x] \_\_call\_\_()
    - [x] class: SubredditListingMixin
        - [x] comments()
        - [x] \_\_init\_\_()
- [x] models/listing/mixins/base.py
    - [x] _prepare()
    - [x] class: BaseListingMixin
        - [x] _validate_time_filter()
        - [x] controversial()
        - [x] hot()
        - [x] new()
        - [x] top()

### CODE REVIEW

**Comment by vikramaditya91** [comment]
@LilSpazJoekp 
Great job on this.

My 2 cents. You could consider making the praw a requirement for asyncprawcore.
This would allow you to simply import non-async related objects directly without having to maintain them.

**Comment by LilSpazJoekp** [comment]
Thank you!

> My 2 cents. You could consider making the praw a requirement for asyncprawcore.
This would allow you to simply import non-async related objects directly without having to maintain them.

Do you mean praw a requirement for asyncpraw? If so, I think that will complicate things (i.e., figuring out what can and can't be imported, it would make docs a mess because you would probably have to reference each other's docs, and most of the examples from the praw docs show synchronous code and would be useless in an async context). Personally, I think that importing in non-async related objects from praw will be more trouble than its worth. I'll see if its feasible to do so. I think it would be possible on some of the mixins.

File: asyncpraw/endpoints.py [hunk 1]
```diff
@@ -0,0 +1,209 @@
+"""List of API endpoints PRAW knows about."""
+
+# flake8: noqa
+# fmt: off
+API_PATH = {
+    "about_edited":            "r/{subreddit}/about/edited/",
+    "about_log":               "r/{subreddit}/about/log/",
```
Comment (vikramaditya91) [inline asyncpraw/endpoints.py:7]: @LilSpazJoekp I had meant that such objects could have been imported from praw.
On one hand it adds to the complexity. On the positive side, you would not have to maintain this yourself

Comment (LilSpazJoekp) [reply asyncpraw/endpoints.py:7]: I don't think it would be hard to maintain. I would apply patches from praw whenever changes were made to it.

Comment (PythonCoderAS) [reply asyncpraw/endpoints.py:7]: One problem can occur whenever API endpoints are modified by PRAW and the changed methods are not backported to asyncpraw. Also models have mixins which have methods that would need to be async-ified.

Comment (LilSpazJoekp) [reply asyncpraw/endpoints.py:7]: Exactly what I was thinking. So I think it would just be easier to do it manually each commit.

**Comment by LilSpazJoekp** [comment]
@PythonCoderAS can you see if you can fix this

**Comment by PythonCoderAS** [comment]
@LilSpazJoekp What's the issue?

**Comment by LilSpazJoekp** [comment]
I tried to merge in praw's commit history and not it won't let me merge

**Comment by PythonCoderAS** [comment]
You need to force push to master. If you changed the root commit, it won't merge in cleanly.

**Comment by LilSpazJoekp** [comment]
I figured it out. I didn't commit the commits from praw when I started working on this. Instead just committed the files.

**Pull request closed without merging**